src/proof/er/er_proof.h

```diff
@@ -0,0 +1,216 @@
+/*********************                                                        */
+/*! \file er_proof.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Alex Ozdemir
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief ER Proof Format
+ **
+ ** Declares C++ types that represent an ER/TRACECHECK proof.
+ ** Defines serialization for these types.
+ **
+ ** You can find details about the way ER is encoded in the TRACECHECK
+ ** format at these locations:
+ **    https://github.com/benjaminkiesl/drat2er
+ **    http://www.cs.utexas.edu/users/marijn/publications/ijcar18.pdf
+ **
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef __CVC4__PROOF__ER__ER_PROOF_H
+#define __CVC4__PROOF__ER__ER_PROOF_H
+
+#include <memory>
+#include <vector>
+
+#include "proof/clause_id.h"
+#include "prop/sat_solver_types.h"
+
+namespace CVC4 {
+namespace proof {
+namespace er {
+
+/**
+ * A definition of the form:
+ *    newVar <-> p v (~x_1 ^ ~x_2 ^ ... ^ ~x_n)
+ */
+struct ErDefinition
+{
+  ErDefinition(prop::SatVariable newVariable,
+               prop::SatLiteral oldLiteral,
+               std::vector<prop::SatLiteral>&& otherLiterals)
+      : d_newVariable(newVariable),
+        d_oldLiteral(oldLiteral),
+        d_otherLiterals(otherLiterals)
+  {
+    // That's all!
+  }
+
+  // newVar
+  prop::SatVariable d_newVariable;
+  // p
+  prop::SatLiteral d_oldLiteral;
+  // A list of the x_i's
+  std::vector<prop::SatLiteral> d_otherLiterals;
+};
+
+// For representing a clause's index within a TRACECHECK proof.
+using TraceCheckIdx = size_t;
+
+/**
+ * A single line in a TRACECHECK proof.
+ *
+ * Consists of a new clause, the index for that clause, and the indices for
```

Comment:
I think that the comma is more accurate here -- the noun phrase is meant to be a list of three items.

Reply:
Oh, it looks like I got a bit confused by the order here :) Maybe it would be slightly better to say "The index of a new clause, the literals of that clause, and the indices..." (to match the order of the arguments and the order in the TRACECHECK format).

Reply:
Good idea!